src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -91,8 +93,15 @@ private void buildServiceMap() {
     addService("KeyFactory", "EC", "EvpKeyFactory$EC");
 
     if (shouldRegisterEdDSA) {
-      addService("KeyFactory", "EdDSA", "EvpKeyFactory$EdDSA");
-      addService("KeyFactory", "Ed25519", "EvpKeyFactory$EdDSA");
+      // KeyFactories are used to convert key encodings to Java Key objects. ACCP's KeyFactory for
+      // Ed25519 returns keys of type EvpEdPublicKey and EvpEdPrivateKey that do not implement
+      // EdECKey interface. One should register the KeyFactories from ACCP if they only use the
+      // output of the factories with ACCP's services.
+      // Once ACCP supports multi-release jar, then this option can be removed.
```

Comment:
Can you elaborate on what you mean here?

Reply:
As the comment says, the EvpEdPublicKey (and EvpEdPrivateKey) does not implement [EdECKey](https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/security/interfaces/EdECKey.html) interfaces, but technically, they should. This interface is not available in JDKs prior to 15. We need change ACCP's build so that on JDK 15+, the EvpEdPublicKey would implement these interfaces. The systematic solution for that is multi-release jar.